pkg/core/db/context_mock.go

```diff
@@ -0,0 +1,40 @@
+package db
```

Comment:
Rename this file to `context_mock_test.go` so it is included in test only.

Reply:
This is also used in tests of hook provider. If it is a test file, the hook provider tests cannot use it.

I think we should use mock_gen for hook provider, then we can do this renaming.